mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/aggregates/UserQuotaThresholds.java

```diff
@@ -41,20 +41,42 @@
 import org.apache.james.util.OptionalUtils;
 
 import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Joiner;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 
 public class UserQuotaThresholds {
 
     public static class Id implements AggregateId {
 
-        public static Id from(User user) {
-            return new Id(user);
+        private static final int PREFIX_INDEX = 0;
+        private static final int NAME_INDEX = 1;
+        private static final int USER_INDEX = 2;
+        private static final String SEPARATOR = "/";
+        private static final String PREFIX = "QuotaThreasholdEvents";
+
+        public static Id fromKey(String key) {
+            List<String> keyParts = Splitter.on(SEPARATOR).splitToList(key);
+            if (keyParts.size() != 3 || !keyParts.get(PREFIX_INDEX).equals(PREFIX)) {
+                throw new IllegalArgumentException();
+            }
+            return new Id(User.fromUsername(keyParts.get(USER_INDEX)), keyParts.get(NAME_INDEX));
+        }
+
+        public static Id from(User user, String name) {
+            return new Id(user, name);
         }
 
         private final User user;
+        private final String name;
 
-        private Id(User user) {
+        private Id(User user, String name) {
+            Preconditions.checkArgument(!user.getLocalPart().contains(SEPARATOR));
```

Comment:
isn't it user.asString() ? What if the domain contains the separator?

Reply:
AFAIK, slash can't be used in a domain

Reply:
How do you explain that I succeeded to write a failing test for this, then?

Reply:
I don't talk about james code but RFC on domain, see https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_hostnames

Reply:
We are testing James, not the RFC ;-)

BTW this might deserve a JIRA ticket, no?